hist/hist/src/TFormula.cxx

```diff
@@ -998,50 +1000,109 @@ void TFormula::FillVecFunctionsShurtCuts() {
 ///    eg.
 ///    varpol2(5) will be replaced with: [5] + [6]*var + [7]*var^2
 ///    Empty name is treated like variable x.
+///    Extended format also supports direct variable specification: pol2(x, 0) or pol(x, [A])
+///    Traditional Syntax: polN - Polynomial of degree of N with default variable x and parameters starting at [0]
+///                        polN(M) - Polynomial of degree N with parameters starting at [M]
+///                        ypolN - Polynomial of degree N with variable y and parameters starting at [0]
+///    Extended Syntax:    polN(var, M) - Polynomial of degree N with variable var and parameters starting at [M]
+///                        polN(var, [A]) - Polynomial of degree N with variable var and parameters starting at [A]
+///    Formula Transformation Examples
+///    Input            Output
+///    pol1              [p0]+[p1]*x
+///    pol2              [p0]+[p1]*x+[p2]*TMath::Sq(x)
+///    ypol1             [p0]+[p1]*y
+///    pol1(x, 0)        [p0]+[p1]*x
+///    pol2(y, 1)        [p1]+[p2]*y+[p3]*TMath::Sq(y)
+///
+///    Extended Code Documentation
+///    The polynomial handling has been enhanced to support explicit variable names
+///    and parameter placeholders in polynomial expressions. This makes the syntax
+///    more consistent with other function types in ROOT (like gaus) and provides
+///    more flexibility in polynomial construction.
+///    The function reports errors through ROOT's error handling system
+///
+///    This documentation update reflects changes made to support
+///    GitHub issue #16794 for enhanced polynomial syntax support.
```

Comment:
This kind of information should be in the commit message.

Reply:
Could you remove this section once it's in the commit message? Users might be confused when they read "Extended Code documentation".
In the doxygen string, we only want to show what formulae are supported. The honours to you for fixing the issue will be in the commit messages and on the issue description. 🙂